pkg/registry/generic/matcher.go

```diff
@@ -93,9 +113,10 @@ type Matcher interface {
 	// include the object's namespace.
 	MatchesSingle() (key string, matchesSingleObject bool)
 
-	// TODO: when we start indexing objects, add something like the below:
-	//         MatchesIndices() (indexName []string, indexValue []string)
-	//       where indexName/indexValue are the same length.
+	// If the index with a given name is supported by the matcher,
+	// returns the value for that index and true, otherwise returns
+	// "" and false.
+	Index(indexName string) (indexValue string, indexSupported bool)
```

Comment:
OK I understand this better now, deleting earlier comment.

Fundamentally the problem this interface is trying to solve is "what index value(s) should this matcher trigger on". I think this interface as written is quickly going to be very limiting. I suggest returning a map or a list of (index name -> value) pairs.

Alternatively, you could define a format for the index values for a particular object, pass that in, and have this return true or false.